tests/models/univnet/test_modeling_univnet.py

```diff
@@ -0,0 +1,359 @@
+    require_torch,
+    require_torch_gpu,
+    slow,
+    torch_device,
+)
+
+from ...test_configuration_common import ConfigTester
+from ...test_modeling_common import (
+    ModelTesterMixin,
+    floats_tensor,
+)
+
+
+if is_torch_available():
+    import torch
+
+    from transformers import UnivNetModel
+
+
+class UnivNetModelTester:
+    def __init__(
+        self,
+        parent,
+        batch_size=2,
+        seq_length=7,
+        in_channels=8,
+        hidden_channels=8,
+        num_mel_bins=20,
+        kernel_predictor_hidden_channels=8,
+        seed=0,
+        is_training=False,
+    ):
+        self.parent = parent
+        self.batch_size = batch_size
+        self.seq_length = seq_length
+        self.in_channels = in_channels
+        self.hidden_channels = hidden_channels
+        self.num_mel_bins = num_mel_bins
+        self.kernel_predictor_hidden_channels = kernel_predictor_hidden_channels
+        self.seed = seed
+        self.is_training = is_training
+
+    def prepare_noise_sequence(self):
+        generator = torch.manual_seed(self.seed)
+        noise_shape = (self.seq_length, self.in_channels)
+        # Create noise on CPU for reproducibility
+        noise_sequence = torch.randn(noise_shape, generator=generator, dtype=torch.float)
+        return noise_sequence
+
+    def prepare_config_and_inputs(self):
+        spectrogram = floats_tensor([self.seq_length, self.num_mel_bins], scale=1.0)
+        noise_sequence = self.prepare_noise_sequence()
+        noise_sequence = noise_sequence.to(spectrogram.device)
+        config = self.get_config()
+        return config, spectrogram, noise_sequence
+
+    def get_config(self):
+        return UnivNetConfig(
+            model_in_channels=self.in_channels,
+            model_hidden_channels=self.hidden_channels,
+            num_mel_bins=self.num_mel_bins,
+            kernel_predictor_hidden_channels=self.kernel_predictor_hidden_channels,
+        )
+
+    def create_and_check_model(self, config, spectrogram, noise_sequence):
+        model = UnivNetModel(config=config).to(torch_device).eval()
+        result = model(spectrogram, noise_sequence)[0]
+        self.parent.assertEqual(result.shape, (1, self.seq_length * 256))
+
+    def prepare_config_and_inputs_for_common(self):
+        config, spectrogram, noise_sequence = self.prepare_config_and_inputs()
+        inputs_dict = {"input_features": spectrogram, "noise_sequence": noise_sequence}
+        return config, inputs_dict
+
+
+@require_torch
+class UnivNetModelTest(ModelTesterMixin, unittest.TestCase):
+    all_model_classes = (UnivNetModel,) if is_torch_available() else ()
+    test_torchscript = False
+    test_pruning = False
+    test_resize_embeddings = False
+    test_resize_position_embeddings = False
+    test_head_masking = False
+    test_mismatched_shapes = False
+    test_missing_keys = False
+    test_model_parallel = False
+    is_encoder_decoder = False
+    has_attentions = False
```

Comment:
Everything is set to false here, could you add a comment as to why we are skipping them?

Reply:
I have added comments with the reasoning for skipping the tests associated with the above flags.

Regarding the `test_torchscript` flag, if I set it to `True`, [`test_torchscript_simple`](https://github.com/huggingface/transformers/blob/e893b1efbbce80b7eaaf24f9e0134450820782b5/tests/test_modeling_common.py#L693) fails due to a `RuntimeError: Only tensors, lists, tuples of tensors, or dictionary of tensors can be output from traced functions` error originating from `torch.jit.trace`:

https://github.com/huggingface/transformers/blob/e893b1efbbce80b7eaaf24f9e0134450820782b5/tests/test_modeling_common.py#L760

I think this test can probably be made to pass (but haven't looked into it closely yet).

[If `test_torchscript` is turned on, some other tests controlled by the `test_torchscript` flag such as [`test_torchscript_output_attentions`](https://github.com/huggingface/transformers/blob/e893b1efbbce80b7eaaf24f9e0134450820782b5/tests/test_modeling_common.py#L698) and [`test_torchscript_output_hidden_state`](https://github.com/huggingface/transformers/blob/e893b1efbbce80b7eaaf24f9e0134450820782b5/tests/test_modeling_common.py#L704) will probably need to be skipped manually since they don't apply to UnivNet.]